operator/slot_ticker/slotticker.go

```diff
@@ -6,83 +6,90 @@ import (
 	"github.com/attestantio/go-eth2-client/spec/phase0"
 )
 
-// The TTicker interface defines a type which can expose a
-// receive-only channel firing slot events.
-type TTicker interface {
-	C() <-chan phase0.Slot
-	Done()
+//go:generate mockgen -package=mocks -destination=./mocks/slotticker.go -source=./slotticker.go
+
+type SlotTickerProvider func() SlotTicker
```

Comment:
Naming wise, I would call this just `Provider` since it will be used in external packages as `slot_ticker.Provider`, and `slot_ticker.SlotTickerProvider` will be really long and repetitive, which isn't needed.